contracts/base/Dispatcher.sol

```diff
@@ -186,7 +193,14 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, V4SwapRout
                             permitSingle := inputs.offset
                         }
                         bytes calldata data = inputs.toBytes(6); // PermitSingle takes first 6 slots (0..5)
-                        PERMIT2.permit(msgSender(), permitSingle, data);
+                        (success, output) = address(PERMIT2).call(
+                            abi.encodeWithSignature(
+                                'permit(address,((address,uint160,uint48,uint48),address,uint256),bytes)',
```

Comment:
auditors: would love to know if theres a better practice as i know `encodeWithSignature` is considered error-prone. `permit` is an overloaded function so we cannot use `encodeCall`

Reply:
Nothing comes to mind, unfortunately.

Reply:
`encodeWithSelector` can't be used for the same reason. Could define a separate interface that uniquely defines the permit functions, but it'll complicate the issue as opposed to just inlining it.